common/http/platform/beast/http_resumer.cpp

```diff
@@ -0,0 +1,454 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/http.hpp>
+
+#include <algorithm>
+
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl/verify_mode.hpp>
+#include <boost/asio.hpp>
+
+#include <common/common.hpp>
+
+namespace mender {
+namespace http {
+
+namespace common = mender::common;
+
+// Represents the parts of a Content-Range HTTP header
+// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range
+struct RangeHeader {
+	long long int range_start;
```

Comment:
I think it is best to leave it signed so that we could handle if a server sends a negative number (?). Also for existing code parsing for example content-length we use `common::StringToLongLong` so it is better to treat them all the same.